### PR TITLE
Make it possible to store server.conf and log.conf as Secret

### PR DESCRIPTION
**Pull request by wiesom**
Because server.conf can contain secrets such as passwords it would be convenient to be able to store the complete server.conf as a Secret instead of a ConfigMap.

With this change it will be possible to the change config store type be setting `questdb.configStorageType=Secret`

### CODE REVIEW

File: charts/questdb/templates/config-map.yaml [hunk 1]
```diff
@@ -1,25 +1,32 @@
 {{- if or .Values.questdb.serverConfig.enabled .Values.questdb.loggingConfig.enabled -}}
 apiVersion: v1
+{{- if eq .Values.questdb.configStorageType "Secret" }}
+kind: Secret
```
Comment (sklarsa) [inline charts/questdb/templates/config-map.yaml:4]: I would prefer if we keep the secret and configmap yaml in separate template files, which seems to be the way that people do things in helm charts these days.  While this does lead to a bit of config duplication, IMO it makes it significantly easier to understand what's going on in each template.

**Review by sklarsa — CHANGES_REQUESTED** [review]

File: charts/questdb/templates/config-map.yaml (hunk 1 above)
Comment (wiesom) [reply charts/questdb/templates/config-map.yaml:4]: Thanks for you input.
The downside I can see with having two different files in this case is that they easily can become out of sync.
By having it in the same file it become obvious that they are mutually exclusive, you can't use both the secret and the config map.

Maybe we could both the secret and the configmap in the same file but not mixed togeheter like it is right now?
Something like this:
```
{{- if eq .Values.questdb.configStorageType "Secret"}}
apiVersion: :v1
kind: Secret
...

{{- else }}
apiVersion: :v1
kind: ConfigMap

{{- end }}
```

FYI. when I created the PR I looked how Grafana Loki solved the same problem
https://github.com/grafana/loki/blob/main/production/helm/loki/templates/config.yaml

Comment (sklarsa) [reply charts/questdb/templates/config-map.yaml:4]: Thanks, you make some good points, and I especially like that grafana example. Let's go in that direction by leaving the file as-is and just renaming it to `config.yaml`. This way there's no confusion about modifying a secret in a template called `config-map.yaml`.

**Review by sklarsa — CHANGES_REQUESTED** [review]

**Review by sklarsa — APPROVED** [review]

**Comment by sklarsa** [comment]
All set, just cut 0.40.7! Thanks for the PRs, and let us know if you have any issues

**Pull request merged**